Bluetooth: Controller: Allow resolving list update during passive scan

cvinayak opened the pull request:
Allow resolving list update  when passive scanning,
otherwise deny if advertising, active scanning, initiating
or periodic sync create is active.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: Controller: Allow resolving list update during passive scan

Allow resolving list update  when passive scanning,
otherwise deny if advertising, active scanning, initiating
or periodic sync create is active.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

wopu-ot commented on subsys/bluetooth/controller/ll_sw/ull_scan.c:
```diff
@@ -617,18 +617,29 @@ uint32_t ull_scan_is_enabled(uint8_t handle)
 {
 	struct ll_scan_set *scan;
 
+	/* NOTE: BIT(0) - passive scanning enabled
+	 *       BIT(1) - active scanning enabled
+	 *       BIT(2) - initiator enabled
+	 *       BIT(3) - periodic sync create active
+	 */
```
These values should really be defines or an enum. That would make the check in ull_filter.c a lot easier to understand.

cvinayak replied:
will rework on this.

That said, these fixes related to HCI are for newer tests, I wish to review the test cases too, just to make sure if they reflect what core spec state.

cvinayak replied:
@wopu-ot please re-review.

cvinayak committed:
Bluetooth: Controller: Use defines for scanning state types

Use defines for scanning state types of passive, active,
initiator and synchronization state.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

Thalley commented on subsys/bluetooth/controller/ll_sw/ull_filter.c:
```diff
@@ -1014,8 +1014,12 @@ static int rl_access_check(bool check_ar)
 	}
 
+	/* NOTE: Allowed when passive scanning, otherwise deny if advertising,
+	 *       active scanning, initiating or periodic sync create is active.
+	 */
 	return ((IS_ENABLED(CONFIG_BT_BROADCASTER) && ull_adv_is_enabled(0)) ||
-		(IS_ENABLED(CONFIG_BT_OBSERVER) && ull_scan_is_enabled(0)))
+		(IS_ENABLED(CONFIG_BT_OBSERVER) &&
+		 (ull_scan_is_enabled(0) & ~ULL_SCAN_IS_PASSIVE)))
 		? 0 : 1;
```
While this looks correct, the double negation makes it hard to read imo. 

Consider doing something like

```
	if (IS_ENABLED(CONFIG_BT_BROADCASTER) && ull_adv_is_enabled(0)) {
		return 0;
	} else if (IS_ENABLED(CONFIG_BT_OBSERVER) &&
		   (ull_scan_is_enabled(0) & ULL_SCAN_IS_PASSIVE) == 0) {
		return 0;
	}

	return 1;
```

Which not only splits the broadcaster check from the observer check, but also makes the check for passive much clearer in my opinion.
